packages/devtools/lib/src/ui/gtags.dart

```diff
@@ -0,0 +1,221 @@
+  external String get app_type;
+  // ignore: non_constant_identifier_names
+  external String get build_type;
+  // ignore: non_constant_identifier_names
+  external String get platform_type;
+}
+
+@JS()
+@anonymous
+class GtagException {
+  external factory GtagException({
+    String description,
+    bool fatal,
+
+    // ignore: non_constant_identifier_names
+    String app_type, // dimension1 (flutter or web)
+    // ignore: non_constant_identifier_names
+    String build_type, // dimension2 (debug or profile)
+    // ignore: non_constant_identifier_names
+    String platform_type, // dimension3 (android or ios)
+  });
+
+  external String get description; // Description of the error.
+  external bool get fatal; // Fatal error.
+  // Custom dimensions:
+  // ignore: non_constant_identifier_names
+  external String get app_type;
+  // ignore: non_constant_identifier_names
+  external String get build_type;
+  // ignore: non_constant_identifier_names
+  external String get platform_type;
+}
+
+/// ****************************************************************************
+/// *** DevTools Property Specific QA events and dimensions.                 ***
+/// ****************************************************************************
+
+// Type of events (event_category):
+const String gaApplicationState = 'application_state'; // type, build, platform
+const String gaScreenViewEvent = 'screen'; // Active screen (tab selected).
+const String gaSelectEvent = 'select'; // User selected something.
+
+const String gaOnError = 'onerror'; // Browser onError detected in DevTools
+const String gaException = 'exception'; // Any Dart exception in DevTools
+
+// DevTools GA screenNames
+const String gaDevTools = 'main';
+const String gaDebugger = 'debugger';
+const String gaInspector = 'inspector';
+const String gaMemory = 'memory';
+const String gaTimeline = 'timeline';
+const String gaLogging = 'loggimng';
+
+// DevTools UI action selected (clicked).
+
+// Main bar UX actions:
+const String gaHotReload = 'hotReload';
+const String gaHotRestart = 'hotRestart';
+const String gaFeedback = 'feedback';
+
+// Common UX actions:
+const String gaPause = 'pause'; // Memory, Timeline, Debugger
+const String gaResume = 'resume'; // Memory, Timeline, Debugger
+
+// Inspector UX actions:
+const String gaWidgetMode = 'widgetMode';
+const String gaRefresh = 'refresh';
+const String gaPerformanceOverlay = 'overlay';
+const String gaDebugPaint = 'debugPaint';
+const String gaPaintBaseline = 'paintBaseline';
+const String gaSlowAnimation = 'slowAnimation';
+const String gaRepaintRainbow = 'repaintRainbow';
+const String gaDebugBanner = 'debugBanner';
+const String gaTrackRebuilds = 'rebuilds';
+const String gaIOS = 'iOS';
+const String gaSelectWidgeMode = 'selectWidgeMode';
+
+// Timeline UX actions:
+const String gaTimelineFrame = 'frame'; // Frame selected in frame chart
+const String gaTimelineFlame = 'flame'; // Select a UI/GPU flame
```

Comment:
now that the cpu profiler is there, could we add one for that? 
const String gaTimelineCpuFlame = 'cpuFlame'; // Select a CPU flame from the CPU profiler.

Reply:
`canvas.onStackFrameSelected.listen` in event_details.dart

Reply:
discussed offline. disregard this suggestion.